enhancements/microshift/dummy-gateway-interface-ovn.md

```diff
@@ -0,0 +1,476 @@
+egressGW and egressIP will not work. A possible solution is to forward those
+traffic to the host kernel, then adding iptables rules to steer the traffic to
+wire.
+
+With this change, the ovn node annotations such
+`k8s.ovn.org/node-primary-ifaddr` and `k8s.ovn.org/l3-gateway-config` will use
+the local IP of br-ex as the node IP, instead of the IP of host's primary
+interface. Therefore the following features that need to read these node
+annotation will be affected. Egress firewall as the egress rules contains the
+node IP. And the traffic case of services with `internalTrafficPolicy=Local`
+backed by hostNetwork pods will break. For services with
+`internalTrafficPolicy=Local`, ovn-k will only put local host endpoints to the
+node switch loadbalancer. If the pod is hostNetwork, the local endpoint means
+the pod with the NodeIP. Normally, the ovn-k will use the nodeIP as the ovn-k
+external IP in node annotation `k8s.ovn.org/l3-gateway-config`. However, in our
+case, as we use a dummy interface for br-ex, we have different IPs for nodeIP
+and ovn-k external IP. So ovn-k cannot identify the right local endpoint in such
+a case. It then put no endpoint to the load-balancer.This problem is not so
+critical to MicroShift. Since we only get one node in the cluster. Creating a
+service with `internalTrafficPolicy=Local` makes no sense in such a case. We can
+fix this problem later until we find a proper solution.
+
+However, the aforementioned features/traffic-case are not required in
+MicroShift. So we want to make the work first and leave advanced features till
+later.
+
+## Design Details
+
+### Open Questions [optional]
+
+1. We'll need to allocate a IP address to br-ex bridge. This IP address will not
+   be exposed externally. Shall we use a fixed IP address, or we allow user to
+   be able to specify it?
+
+   A: We decide to reuse the `advertise-address` of kubernetes API server by add
+   this address to br-ex. Users can specify the address with the config flag
+   `apiServer.advertiseAddress` in microshift. The default value is `10.44.0.0`.
+
+### Test Plan
+
+The current tests we have are sufficient for testing this networking change.
+
+### Graduation Criteria
+
+N/A
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+N/A
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+When doing upgrade, the microshift-ovs-init.service will be responsible for
+deleting the br-ex bridge and NetworkManager config files that is created by
+previous version with NetworkManager.
+
+Downgrade cannot be done automatically, users will need to manually remove the
+dummy interface and br-ex bridge created by microshift-ovs-init.service.
+
+### Version Skew Strategy
+
+N/A
+
+### Operational Aspects of API Extensions
+
+N/A
+
+#### Failure Modes
+
+N/A
+
+#### Support Procedures
+
+N/A
+
+## Implementation History
+
+N/A
+
+## Alternatives
```

Comment:
Do either of these alternatives avoid any of the drawbacks listed above?

Reply:
No. Furthermore,  the disabled gateway approach may have more drawbacks, as it drops all the gateway elements.